src/schedule/dma_multi_chan_domain.c

```diff
@@ -303,11 +304,23 @@ static bool dma_multi_chan_domain_is_pending(struct ll_schedule_domain *domain,
 			    pipe_task->sched_comp)
 				continue;
 
-			/* it's too soon for this task */
-			if (!pipe_task->registrable &&
-			    pipe_task->task.start >
-			    platform_timer_get_noirq(timer_get()))
-				continue;
+			if (domain->full_sync) {
```

Comment:
Can you put a comment here saying what full_sync means and why the action needs to be done differently.